### DISCUSSION TITLE
_rsc requests even while navigating between client components.

### DISCUSSION THREAD

**Discussion by ImanAkrami** [vercel/next.js#60951] [Help]
### Summary

Hi
Maybe I don't understand something, but when i navigate between two client component routes, there is still a request to _rsc?

**Comment by umairb-voicevox** [comment]
Documentation on this is really poor. These phantom `_rsc` requests are extremely frustrating.

**Reply by kumar0** [reply]
Any workaround you found ?

**Comment by rajeshdavidbabu** [comment]
The answer is that even if you place "use client" at the very top of your component tree, including your root layout.tsx, the Next.js server is still fundamentally involved in every navigation, and an RSC payload will still be requested.

The App Router's Core Philosophy:  Server-Driven Navigation

The Next.js App Router is not a traditional Single-Page Application (SPA) router that operates exclusively in the browser. It is a server-centric router that uses React Server Components as its underlying primitive.

Think of it this way: the "use client" directive does not turn off the RSC architecture. Instead, it acts as a boundary marker that tells the server: "Stop rendering on the server at this point. Instead, send a placeholder and a reference to a JavaScript bundle that the client should download and execute."

- Here's what happens on a page switch, even if the destination page and layout are marked with "use client":
   - Navigation is Intercepted: When you click a <Link> component, the App Router intercepts the navigation event. It does not immediately try to render the new page on the client.
   - A Request is Made to the Server: The router's first action is always to make a request to the Next.js server for the new route.
   - The Server Generates the "Blueprint" (RSC Payload): The server receives the request and processes it. Because your entire page is a Client Component, the server's job is simple but essential. It generates an RSC payload that is effectively a set of instructions for the client-side router.
